public/app/features/plugins/components/AppRootPage.test.tsx

```diff
@@ -106,23 +111,150 @@ describe('AppRootPage', () => {
 
     importAppPluginMock.mockResolvedValue(plugin);
 
-    renderUnderRouter();
-
+    // Renders once for the first time
+    await renderUnderRouter();
     expect(await screen.findByText('my great component')).toBeVisible();
-
-    // renders the first time
     expect(RootComponent.timesRendered).toEqual(1);
 
+    // Does not render again when navigating to a non-plugin path
     await act(async () => {
       locationService.push('/foo');
     });
-
     expect(RootComponent.timesRendered).toEqual(1);
 
+    // Renders it again when navigating back to a plugin path
     await act(async () => {
       locationService.push('/a/my-awesome-plugin');
     });
-
     expect(RootComponent.timesRendered).toEqual(2);
   });
+
+  describe('When accessing using different roles', () => {
+    beforeEach(() => {
+      const pluginMetaWithIncludes = getMockPlugin({
+        id: 'my-awesome-plugin',
+        type: PluginType.app,
+        enabled: true,
+        includes: [
+          {
+            type: PluginIncludeType.page,
+            name: 'Awesome page 1',
+            path: '/a/my-awesome-plugin/viewer-page',
+            role: 'Viewer',
+          },
+          {
+            type: PluginIncludeType.page,
+            name: 'Awesome page 2',
+            path: '/a/my-awesome-plugin/editor-page',
+            role: 'Editor',
+          },
+          {
+            type: PluginIncludeType.page,
+            name: 'Awesome page 2',
+            path: '/a/my-awesome-plugin/admin-page',
+            role: 'Admin',
+          },
+          {
+            type: PluginIncludeType.page,
+            name: 'Awesome page 2',
+            path: '/a/my-awesome-plugin/page-without-role',
+          },
+        ],
+      });
+
+      getPluginSettingsMock.mockResolvedValue(pluginMetaWithIncludes);
+
+      const plugin = new AppPlugin();
+      plugin.meta = pluginMetaWithIncludes;
+      plugin.root = RootComponent;
+
+      importAppPluginMock.mockResolvedValue(plugin);
+    });
+
+    it('an User with no existing role should be able to see any page', async () => {
```

Comment:
Should this be `an User with no existing role should NOT be able to see any page` ?

Reply:
you are right!